src/AdoPat/PatOptions.cs

```diff
@@ -33,9 +36,23 @@ public record PatOptions
         /// <returns>The cache key.</returns>
         public string CacheKey()
         {
-            var strings = new List<string> { this.Organization, this.DisplayName };
-            strings.AddRange(this.Scopes);
-            return string.Join(" ", strings);
+            // We concatenate scopes with the empty string after sorting and
+            // deduplicating them to ensure the cache key is always the same.
+            var sortedScopes = string.Concat(this.Scopes.ToImmutableSortedSet());

```

Comment:
This is doing both sorting and deduplication, correct? So we have an empty string as a scope then it will sort with the empty string first, correct?

Reply:
Yes, `ToImmutableSortedSet` is a LINQ method. I've also asked about scope validation, so I'd be happy to see handling empty string scopes in a follow up PR about scope validation.

Reply:
I _think_ that's the case, but also we probably just shouldn't allow the empty string to be a scope.